crates/trigger/src/runtime_config/key_value.rs

```diff
@@ -122,19 +124,33 @@ impl RedisKeyValueStoreOpts {
 
 #[derive(Clone, Debug, Deserialize)]
 pub struct AzureCosmosConfig {
+    #[serde(default)]
     key: String,
     account: String,
     database: String,
     container: String,
+    #[serde(default)]
+    environmental_auth: bool,
 }
 
 impl AzureCosmosConfig {
     pub fn build_store(&self) -> Result<Arc<dyn StoreManager>> {
+        let auth_options = if self.key.is_empty() || self.environmental_auth {
+            KeyValueAzureCosmosAuthOptions::Environmental
+        } else {
+            if self.key.is_empty() {
```

Comment:
I am not sure how this could happen.  Wouldn't this condition always go down the "true" clause of the `if`?

Reply:
Yes, 🤣 that was left over from moving some code around.